app/models/handlers/datadog_event.rb

```diff
@@ -26,10 +26,8 @@ class DatadogEvent < Base
       example: 'normal'
 
     def call
-      return if [
-        api_key,
-        metric_name
-      ].any?(&:blank?)
+      raise(UserError, "Missing 'api_key'") if api_key.blank?
+      raise(UserError, "Missing 'metric_name'") if metric_name.blank?
```

Comment:
I converted these early returns as-is, but I think we can leverage the settings and for anything that is not flagged as "optional" to automatically run this logic.

We need to have "runtime validations" so we can catch invalid variable replacement to hit the handler code which could lead to error messges that would be hard to understand.

eg.: `"{{ custommer_with_a_typo }}"` => `""`